packages/openai-compatible/src/openai-compatible-chat-language-model.ts

```diff
@@ -0,0 +1,537 @@
+import {
+  InvalidResponseDataError,
+  LanguageModelV1,
+  LanguageModelV1CallWarning,
+  LanguageModelV1FinishReason,
+  LanguageModelV1ProviderMetadata,
+  LanguageModelV1StreamPart,
+  UnsupportedFunctionalityError,
+} from '@ai-sdk/provider';
+import {
+  FetchFunction,
+  ParseResult,
+  combineHeaders,
+  createEventSourceResponseHandler,
+  createJsonResponseHandler,
+  generateId,
+  isParsableJson,
+  postJsonToApi,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod';
+import { convertToOpenAICompatibleChatMessages } from './convert-to-openai-compatible-chat-messages';
+import { getResponseMetadata } from './get-response-metadata';
+import {
+  OpenAICompatibleChatModelId,
+  OpenAICompatibleChatSettings,
+} from './openai-compatible-chat-settings';
+import {
+  openAICompatibleErrorDataSchema,
+  openAICompatibleFailedResponseHandler,
+} from './openai-compatible-error';
+import { prepareTools } from './openai-compatible-prepare-tools';
+import { mapOpenAICompatibleFinishReason } from './map-openai-compatible-finish-reason';
+
+type OpenAICompatibleChatConfig = {
+  provider: string;
+  headers: () => Record<string, string | undefined>;
+  url: (options: { modelId: string; path: string }) => string;
+  fetch?: FetchFunction;
+};
+
+export class OpenAICompatibleChatLanguageModel implements LanguageModelV1 {
+  readonly specificationVersion = 'v1';
+
+  readonly supportsStructuredOutputs = false;
+
+  readonly modelId: OpenAICompatibleChatModelId;
+  readonly settings: OpenAICompatibleChatSettings;
+
+  private readonly config: OpenAICompatibleChatConfig;
+
+  constructor(
+    modelId: OpenAICompatibleChatModelId,
+    settings: OpenAICompatibleChatSettings,
+    config: OpenAICompatibleChatConfig,
```

Comment:
include the default object generation mode in `config`

Reply:
I'd like to define a type for the values `'json' | 'tool' | undefined`. It appears the first reference is in LanguageModelV1. Is it appropriate to add something [there](https://github.com/vercel/ai/blob/a334c6ad18ce13eff0e5072783c902a93754f0fe/packages/provider/src/language-model/v1/language-model-v1.ts#L39)?